forte/mcscf/mcscf_orb_grad.cc

```diff
@@ -475,35 +494,69 @@ void MCSCF_ORB_GRAD::fill_tei_custom(ambit::BlockedTensor V) {
     }
 }
 
-void MCSCF_ORB_GRAD::build_tei_from_ao() {
+void MCSCF_ORB_GRAD::build_tei_df() {
     if (nactv_ == 0)
         return;
 
-    // This function will do an integral transformation using the JK builder,
-    // and return the integrals of type <px|uy> = (pu|xy).
     timer_on("Build (pu|xy) integrals");
 
-    // Transform C matrix to C1 symmetry
-    // JK does not support mixed symmetry needed for 4-index integrals (York 09/09/2020)
-    auto aotoso = ints_->wfn()->aotoso();
-    auto C_nosym = std::make_shared<psi::Matrix>(nso_, nmo_);
-
-    // Transform from the SO to the AO basis for the C matrix
-    // MO in Pitzer ordering and only keep the non-frozen MOs
-    for (int h = 0, index = 0; h < nirrep_; ++h) {
-        for (int i = 0; i < nmopi_[h]; ++i) {
-            int nao = nso_, nso_h = nsopi_[h];
+    auto C_nosym = ints_->Ca_SO2AO(C_);
 
-            if (!nso_h)
-                continue;
+    auto Cact = std::make_shared<psi::Matrix>("Cact", nso_, nactv_);
```

Comment:
I think `get_column` better convey the point here: grab coefficients for active orbitals. Something like `get_block(Slice({0}, {nso_}), Slice({ndocc}, {ndocc+nactv}))` looks a bit vague for the the row slice because it reflects AO index instead of MO index.

Reply:
That's fair, especially given that `actv_mos` is a list... Slices are made for continuous access, lists for random.